### PR TITLE
feat: add config az func secret store extension

### LINKED ISSUE #334: Add missing extension for registering secret store in Azure Functions
**Is your feature request related to a problem? Please describe.**
Currently, we have an `.ConfigureSecretStore` extension for Azure Functions to register the secret store, but when using the `IConfiguration` and/or the context (environment information), one still has to call and retrieve it themselves via `IFunctionsHostBuilder.GetContext`.

**Describe the solution you'd like**
As we provide an extension overload in general applications with the `IConfiguration`, we should consider to streamline this in Azure Functions as well. We can use the `FunctionsHostBuilderContext` and `IConfiguration` besides the `SecretStoreBuilder` in the overload.

### PR DESCRIPTION
**Pull request by stijnmoreels**
Adds an `.ConfigureSecretStore` to the Azure Functions environment that has access to the `IConfiguration` and `IFunctionsHostContext`. This extra method overload will streamline the secret store registration with the already available extensions on `IHostBuilder`.

Closes #334

### CODE REVIEW

File: src/Arcus.Security.AzureFunctions/Extensions/IFunctionHostBuilderExtensions.cs [hunk 1]
```diff
@@ -27,5 +28,23 @@ public static IFunctionsHostBuilder ConfigureSecretStore(this IFunctionsHostBuil
...
+        /// <param name="configureSecretStores">The customization of the different target secret store sources to include in the final <see cref="ISecretProvider"/>.</param>
+        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="functionsHostBuilder"/> or <paramref name="configureSecretStores"/> is <c>null</c>.</exception>
+        public static IFunctionsHostBuilder ConfigureSecretStore(
+            this IFunctionsHostBuilder functionsHostBuilder,
+            Action<FunctionsHostBuilderContext, IConfiguration, SecretStoreBuilder> configureSecretStores)
+        {
+            Guard.NotNull(functionsHostBuilder, nameof(functionsHostBuilder), "Requires a functions host builder to add the secret store");
+            Guard.NotNull(configureSecretStores, nameof(configureSecretStores), "Requires a function to configure the secret store with potential secret providers");
+
+            FunctionsHostBuilderContext context = functionsHostBuilder.GetContext();
```
Comment (stijnmoreels) [inline src/Arcus.Security.AzureFunctions/Extensions/IFunctionHostBuilderExtensions.cs:45]: Not sure if this should be called here or inside the `AddSecretStore`. The call would be delayed, and outside of the context. I didn't see much context-specifics when inspecting `.GetContext` but not sure if we should take the risk of moving it inside the lambda.

Comment (fgheysels) [reply src/Arcus.Security.AzureFunctions/Extensions/IFunctionHostBuilderExtensions.cs:45]: I would keep it as is.

File: docs/preview/features/secret-store/azure-functions.md [hunk 2]
```diff
@@ -28,8 +28,7 @@ namespace MyHttpAzureFunction
     {
         public override void Configure(IFunctionsHostBuilder builder)
         {
-            IConfiguration config = builder.GetContext().Configuration;
-            builder.ConfigureSecretStore(stores =>
+            builder.ConfigureSecretStore((FunctionsHostBuilderContext context, IConfiguration config, SecretStoreBuilder stores) =>
```
Comment (fgheysels) [inline docs/preview/features/secret-store/azure-functions.md:31]: Technically , this is a breaking change...

We must also make sure that this change i reflected in our docs / code samples.

**Review by fgheysels — APPROVED** [review]

File: docs/preview/features/secret-store/azure-functions.md (hunk 2 above)
Comment (stijnmoreels) [reply docs/preview/features/secret-store/azure-functions.md:31]: Ah, I think not, bc we have only added an overload. So you could still use the one from before, it's just that there's now a new one that is 'better' so we could update the docs.

Comment (fgheysels) [reply docs/preview/features/secret-store/azure-functions.md:31]: Shall we update the documentation in this PR as well ?

Comment (stijnmoreels) [reply docs/preview/features/secret-store/azure-functions.md:31]: The documentation is already updated, this comment is even on a updated preview docs. 😉

Comment (fgheysels) [reply docs/preview/features/secret-store/azure-functions.md:31]: I need to pay more attention :D

**Review by fgheysels — APPROVED** [review]

**Pull request merged**